api/v1alpha3/azurecluster_conversion.go

```diff
@@ -215,7 +215,7 @@ func Convert_v1alpha4_SecurityGroup_To_v1alpha3_SecurityGroup(in *infrav1alpha4.
 	out.IngressRules = make(IngressRules, 0)
 	for i, rule := range in.SecurityRules {
 		if rule.Direction == infrav1alpha4.SecurityRuleDirectionInbound { // only inbound rules are supported in v1alpha3.
-			out.IngressRules[i] = IngressRule{}
+			out.IngressRules = append(out.IngressRules, IngressRule{})
 			if err := Convert_v1alpha4_SecurityRule_To_v1alpha3_IngressRule(&in.SecurityRules[i], &out.IngressRules[i], s); err != nil {
```

Comment:
won't out.IngressRules[i] also cause issues? I think we need a counter for out.IngressRules

Reply:
Yep, that was silly. I've updated. PTAL